src/realm/db.cpp

```diff
@@ -1097,32 +1098,46 @@ void DB::do_open(const std::string& path, bool no_create_file, bool is_backend,
                         good_history_type = (stored_hist_type == Replication::hist_InRealm ||
                                              stored_hist_type == Replication::hist_None);
                         if (!good_history_type)
-                            throw IncompatibleHistories("Expected a Realm with no or in-realm history", path);
+                            throw IncompatibleHistories(
+                                util::format(
+                                    "Expected a Realm with no or in-realm history, but found history type %1",
+                                    stored_hist_type),
+                                path);
                         break;
                     case Replication::hist_SyncClient:
                         good_history_type = ((stored_hist_type == Replication::hist_SyncClient) || (top_ref == 0));
                         if (!good_history_type)
                             throw IncompatibleHistories(
-                                "Expected an empty Realm or a Realm written by Realm Mobile Platform", path);
+                                util::format("Expected an empty Realm or a Realm written by Realm Mobile Platform, "
```

Comment:
"Realm Mobile Platform" should likely be updated.
Perhaps "Expected an empty or synced Realm"?

Reply:
done. @kraenhansen, any issue with this?